src/lib/services/proxy-service.ts

```diff
@@ -45,21 +47,31 @@ export class ProxyService {
 
     private readonly services: Services;
 
+    private readonly globalFrontendApiCache: GlobalFrontendApiCache;
+
     /**
-     * This is intentionally a Promise becasue we want to be able to await
+     * This is intentionally a Promise because we want to be able to await
      * until the client (which might be being created by a different request) is ready
      * Check this test that fails if we don't use a Promise: src/test/e2e/api/proxy/proxy.concurrency.e2e.test.ts
      */
     private readonly clients: Map<ApiUser['secret'], Promise<Unleash>> =
         new Map();
+    private readonly newClients: Map<ApiUser['secret'], Promise<Unleash>> =
```

Comment:
we store new and old clients side by side. when flag is disabled new clients will be empty.